tools/esdlcmd/esdlcmd_core.cpp

```diff
@@ -1167,13 +1168,166 @@ class Esdl2CppCmd : public EsdlHelperConvertCmd
+        if (iter.matchOption(optOutDirPath, ESDL_CONVERT_OUTDIR))
+            return true;
+        if (iter.matchFlag(optRollup, ESDL_OPTION_ROLLUP))
+            return true;
+        if (iter.matchFlag(optProcessAllIncludes, ESDLOPT_RECURSIVE) || iter.matchFlag(optProcessAllIncludes, ESDLOPT_RECURSIVE_S))
+            return true;
+        if (EsdlConvertCmd::parseCommandLineOption(iter))
+            return true;
+
+        return false;
+    }
+
+    esdlCmdOptionMatchIndicator matchCommandLineOption(ArgvIterator &iter, bool finalAttempt)
+    {
+        return EsdlConvertCmd::matchCommandLineOption(iter, true);
+    }
+
+    virtual bool finalizeOptions(IProperties *globals) override
+    {
+        extractEsdlCmdOption(optIncludePath, globals, ESDLOPT_INCLUDE_PATH_ENV, ESDLOPT_INCLUDE_PATH_INI, NULL, NULL);
+
+        if (optSource.isEmpty())
+        {
+            usage();
+            throw( MakeStringException(0, "\nError: Source file parameter required\n"));
+        }
+
+        if (optRollup)
+        {
+            if(optService.isEmpty())
+            {
+                usage();
+                throw( MakeStringException(0, "\nError: `--service` option required when generating rolled-up output\n"));
+            }
+        }
+
+        cmdHelper.verbose = optVerbose;
+        return true;
+    }
+
+    virtual void doTransform(IEsdlDefObjectIterator& objs, StringBuffer &out, double version=0, IProperties *opts=NULL, const char *ns=NULL, unsigned flags=0 ) override
+    {
+    }
+
+    virtual void loadTransform( StringBuffer &xsltpath, IProperties *params ) override
+    {
+    }
+
+    virtual void setTransformParams(IProperties *params ) override
+    {
+    }
+
+    virtual int processCMD() override
+    {
+        if (!optRollup)
+        {
+            // duplicate behavior of the old esdl2xml command
+            Esdl2Esxdl old2XmlCmd(optProcessAllIncludes, optVerbose);
+            old2XmlCmd.transform(optSource, optOutDirPath);
+        } else {
+            StringBuffer esxml;
+            cmdHelper.getServiceESXDL(optSource.get(), optService.get(), esxml, 0, NULL, (DEPFLAG_INCLUDE_TYPES & ~DEPFLAG_INCLUDE_METHOD), optIncludePath.str(), optTraceFlags());
+            if (esxml.length()==0)
+            {
+                fprintf(stderr,"\nESDL Definition for service %s could not be loaded from: %s\n", optService.get(), optSource.get());
+                return -1;
+            }
+            StringBuffer filename;
+            splitFilename(optSource.str(), nullptr, nullptr, &filename, nullptr);
+            saveAsFile(optOutDirPath.get(), filename.str(), esxml.str(), ".xml");
+        }
+        return 0;
+    }
+
+    void printOptions()
+    {
+        puts("Options:");
+        puts("   --rollup         Output a single XML file containing all definitions used by");
+        puts("                    <serviceName>");
+        puts("   -r, --recursive  Process all includes. Implied when generating monolithic");
```

Comment:
I see the help is correct for the short version with a single dash.  The define must have been a typo.

Reply:
Yes I think so.